plugins/experimental/uri_signing/uri_signing.c

```diff
@@ -218,6 +224,32 @@ TSRemapDoRemap(void *ih, TSHttpTxn txnp, TSRemapRequestInfo *rri)
       checkpoints[cpi++] = mark_timer(&t);
     }
     jws = get_jws_from_cookie(&client_cookie, &client_cookie_sz_ct, package);
+  } else {
+    /* There has been a JWS found in the url */
+    /* Strip the token from the URL for upstream if configured to do so */
+    if (config_strip_token((struct config *)ih)) {
+      if (strncmp(url, strip_uri, strip_ct) != 0) {
+        char *strip_uri_start = &strip_uri[0];
+        char *strip_uri_end   = &strip_uri[strip_ct];
+        PluginDebug("Stripping token from upstream url to: %s", strip_uri_start);
+        if (TSUrlParse(rri->requestBufp, rri->requestUrl, (const char **)&strip_uri_start, strip_uri_end) != TS_PARSE_DONE) {
+          PluginDebug("Error in TSUrlParse");
+        } else {
+          status = TSREMAP_DID_REMAP;
+        }
+      }
+    }
+
+    /* Check auth_dir and pass through if configured */
+    if (uri_matches_auth_directive((struct config *)ih, url, url_ct)) {
+      if (url != NULL) {
+        TSfree((void *)url);
+      }
+      if (url != NULL) {
```

Comment:
This is probably technically correct, but maybe by accident? Should this say `strip_uri != NULL`?

Reply:
fixed